searx/engines/nixpkgs.py

```diff
@@ -0,0 +1,171 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+""".. sidebar:: info
+
+   - :origin:`nixpkgs.py <searx/engines/nixpkgs.py>`
+   - `Nixpkgs Search <https://search.nixos.org>`_
+
+nixpkgs_ makes it easy to search the huge nixpkgs package index.
+The search backend on search.nixos.org is an Elasticsearch instance.
+
+Example
+=======
+
+The following is an example configuration for an nixpkgs_ engine with
+authentication configured.
+
+Credentials are available here: https://github.com/NixOS/nixos-search/blob/main/frontend/src/index.js
+
+.. code:: yaml
+
+  - name: nixpkgs
+    shortcut: nix
+    engine: nixpkgs
+    base_url: https://search.nixos.org/backend
+    username: 
+    password: 
+    index: latest-42-nixos-unstable
+    channel: unstable
+
+"""
+
+# about
+about = {
+    "website": "https://search.nixos.org",
+    "use_official_api": False,
+    "require_api_key": True,
+}
+
+categories = ["it", "packages"]
+
+from json import loads
+from searx.exceptions import SearxEngineAPIException
+
+base_url = "https://search.nixos.org/backend"
+username = ""
+password = ""
+index = "latest-42-nixos-unstable"
+channel = "unstable"
+search_url = f"{base_url}/{index}/_search"
+show_metadata = False
+categories = ["it", "packages"]
+
+
+def init(engine_settings):
+    if not (username and password):
+        raise ValueError("username and password need to be configured.")
+
+
+def request(query, params):
+    params["auth"] = (username, password)
+    params["url"] = search_url
+    params["method"] = "GET"
```

Comment:
This can be dropped as GET is the default anyways.

Reply:
Alright, thanks! Fixed it :)